reference_designs/ipu-xrt/matrix_multiplication_column/aie2.py

```diff
@@ -0,0 +1,246 @@
+#
+# This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# (c) Copyright 2023 AMD Inc.
+
+import aie
+from aie.ir import *
+from aie.dialects.func import *
+from aie.dialects.scf import *
+from aie.dialects.aie import *
+from aie.dialects.aiex import *
+from aie.dialects.extras import memref, arith
+from aie.util import mlir_mod_ctx
+
+
+def my_matmul():
+    M = 256
+    K = 128
+    N = 128
+    m = 64
+    k = 32
+    n = 64
+    r = 4
+    s = 4
+    t = 4
+    word_size_in = 2
+    word_size_out = 2
+
+    n_cores = 4
```

Comment:
It could be cool to make these variables (line 19 - 31) command line arguments to the script down the line at some point to make it easy to experiment with different sizes.

Reply:
This would also require the C kernel to be generated with different tile sizes (or support a select set of sizes that are always generated), so maybe out of scope.